cmd/podman/main_local.go

```diff
@@ -25,8 +26,24 @@ import (
 const remote = false
 
 func init() {
-
-	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.CGroupManager, "cgroup-manager", "", "Cgroup manager to use (cgroupfs or systemd, default systemd)")
+	cgroupManager := libpod.SystemdCgroupsManager
+	if runtimeConfig, err := libpod.DefaultRuntimeConfig(); err == nil {
+		cgroupManager = runtimeConfig.CgroupManager
+	}
+	cgroupHelp := "Cgroup manager to use (cgroupfs or systemd)"
+	if rootless.IsRootless() {
+		cgroupv2, _ := util.IsCgroup2UnifiedMode()
+		if cgroupv2 {
+			if cgroupManager != libpod.SystemdCgroupsManager {
+				logrus.Warnf("cgroup_manager must be %s, ignoring configuration %s.", libpod.SystemdCgroupsManager, cgroupManager)
```

Comment:
I think with cgroup v2 both `cgroupfs` and `systemd` are valid so we don't need to override it